app/pages/patientdata/patientdata.js

```diff
@@ -124,7 +114,7 @@ export let PatientData = React.createClass({
   },
 
   renderPatientData: function() {
-    if (this.props.fetchingPatient || this.props.fetchingPatientData) {
+    if (this.props.fetchingPatient || this.props.fetchingPatientData || this.state.processingData) {
```

Comment:
Ah, so I see you won the bet on this. The GIF still freezes when you hit the data processing step. I guess the only thing that could change that is putting the data processing in a Web Worker, which is on the roadmap at least for investigation for future-tideline.